src/Course/Traversable.hs

```diff
@@ -34,3 +38,77 @@ instance Traversable List where
     -> f (List b)
   traverse f =
     foldRight (\a b -> (:.) <$> f a <*> b) (pure Nil)
+
+instance Traversable ExactlyOne where
+  traverse ::
+    Applicative f =>
+    (a -> f b)
+    -> ExactlyOne a
+    -> f (ExactlyOne b)
+  traverse =
+    error "todo: Course.Traversable traverse#instance ExactlyOne"
+
+instance Traversable Optional where
+  traverse ::
+    Applicative f =>
+    (a -> f b)
+    -> Optional a
+    -> f (Optional b)
+  traverse =
+    error "todo: Course.Traversable traverse#instance Optional"
+
+-- | Sequences a traversable value of structures to a structure of a traversable value.
+--
+-- >>> sequenceA (ExactlyOne 7 :. ExactlyOne 8 :. ExactlyOne 9 :. Nil)
+-- ExactlyOne [7,8,9]
+--
+-- >>> sequenceA (Full (ExactlyOne 7))
+-- ExactlyOne (Full 7)
+--
+-- >>> sequenceA (Full (*10)) 6
+-- Full 60
+sequenceA ::
+  (Applicative f, Traversable t) =>
+  t (f a)
+  -> f (t a)
+sequenceA =
+  error "todo: Course.Traversable#sequenceA"
+
+instance (Traversable f, Traversable g) =>
+  Traversable (Compose f g) where
+-- Implement the traverse function for a Traversable instance for Compose
+  traverse =
+    error "todo: Course.Traversable traverse#instance (Compose f g)"
+
+-- | The `Product` data type contains 2 values from both type constructors.
```

Comment:
It might be clearer to word this "The `Product` data type contains one value from each of the two type constructors."